datastore/complete_post_object.go

```diff
@@ -0,0 +1,86 @@
+package datastore
+
+import (
+	"fmt"
+
+	nex "github.com/PretendoNetwork/nex-go"
+	common_globals "github.com/PretendoNetwork/nex-protocols-common-go/globals"
+	datastore "github.com/PretendoNetwork/nex-protocols-go/datastore"
+	datastore_types "github.com/PretendoNetwork/nex-protocols-go/datastore/types"
+)
+
+func completePostObject(err error, client *nex.Client, callID uint32, param *datastore_types.DataStoreCompletePostParam) uint32 {
+	if commonDataStoreProtocol.s3ObjectSizeHandler == nil {
+		common_globals.Logger.Warning("S3ObjectSize not defined")
+		return nex.Errors.Core.NotImplemented
+	}
+
+	if commonDataStoreProtocol.getObjectSizeDataIDHandler == nil {
+		common_globals.Logger.Warning("GetObjectSizeDataID not defined")
+		return nex.Errors.Core.NotImplemented
+	}
+
+	if commonDataStoreProtocol.updateObjectUploadCompletedByDataIDHandler == nil {
+		common_globals.Logger.Warning("UpdateObjectUploadCompletedByDataID not defined")
+		return nex.Errors.Core.NotImplemented
+	}
+
+	if err != nil {
+		common_globals.Logger.Error(err.Error())
+		return nex.Errors.DataStore.Unknown
+	}
+
+	// TODO - What is param.IsSuccess? Is this correct?
```

Comment:
Are there any games which actually behave this way? Super Mario Maker simply throws up an error when the S3 upload fails. I don't recall what Tipping Stars does in these cases, and SM3DW doesn't post objects. Those are the only games I've worked on that use DataStore

If that's how the client is supposed to behave though then I believe the implementation is correct no? The comment just needs to be removed?

Reply:
In Badge Arcade if the upload fails, it will send the `CompletePostObject` without the `isSuccess` param set, and then show an error to the user.

> If that's how the client is supposed to behave though then I believe the implementation is correct no? The comment just needs to be removed?

Yes, this should be correct, although I think we should remove the data ID from the server if the operation fails (this is after checking that the data ID actually doesn't exist on S3)

Reply:
Love that SMM just doesn't do that lol. Okay sounds good